pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -435,6 +450,22 @@ func isHealthy(services []common.ApplicationComponentStatus) bool {
 func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 	// If Application Own these two child objects, AC status change will notify application controller and recursively update AC again, and trigger application event again...
 	return ctrl.NewControllerManagedBy(mgr).
+		Watches(&source.Kind{
+			Type: &v1beta1.ResourceTracker{},
+		}, ctrlHandler.Funcs{
+			CreateFunc: func(createEvent ctrlEvent.CreateEvent, limitingInterface workqueue.RateLimitingInterface) {
+				handleResourceTracker(createEvent.Object, limitingInterface)
+			},
+			UpdateFunc: func(updateEvent ctrlEvent.UpdateEvent, limitingInterface workqueue.RateLimitingInterface) {
+				handleResourceTracker(updateEvent.ObjectNew, limitingInterface)
+			},
+			DeleteFunc: func(deleteEvent ctrlEvent.DeleteEvent, limitingInterface workqueue.RateLimitingInterface) {
+				handleResourceTracker(deleteEvent.Object, limitingInterface)
+			},
+			GenericFunc: func(genericEvent ctrlEvent.GenericEvent, limitingInterface workqueue.RateLimitingInterface) {
+				handleResourceTracker(genericEvent.Object, limitingInterface)
+			},
+		}).
```

Comment:
what are these handler for?

Reply:
These handlers are added for application controller to response ResourceTracker change events. Designed by @leejanee .

Reply:
What's the real affect? Who will change RT besides app controller?

Reply:
In the mode of KeepLegacyResource, history versioned ResourceTrackers will not be automatically deleted by application controller. Instead, users have the control to select which history version to keep or discard. In this case, user will delete RT.

Reply:
I got it. That means app controller is also the controller of RT